catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/inputtransformer/InputTransformerProducer.java

```diff
@@ -71,56 +72,63 @@ protected Object transform(
     }
 
     MimeType derivedMimeType = null;
-    if (StringUtils.isEmpty(mimeType)) {
-      try (TemporaryFileBackedOutputStream fileBackedOutputStream =
-          new TemporaryFileBackedOutputStream()) {
-        try {
-          IOUtils.copy(message, fileBackedOutputStream);
-          derivedMimeType =
-              new MimeType(
-                  getMimeTypeFromHeader(in, fileBackedOutputStream.asByteSource().openStream()));
-        } catch (IOException e) {
-          LOGGER.debug("Failed to copy incoming inputStream message", e);
-        } finally {
-          message = fileBackedOutputStream.asByteSource().openStream();
+    try (TemporaryFileBackedOutputStream tfbos = new TemporaryFileBackedOutputStream()) {
```

Comment:
✏️ We could declare `message` in the try-with-resources, then move the null check inside and delete the finally block.